app/controllers/api/repositories_controller.rb

```diff
@@ -64,8 +64,15 @@ def sync
 
   private
 
+  # handle a missing owner or name parameter when doing the lookup for Repository in find_repo
+  # and return a formatted JSON with the missing parameter name with a 400 HTTP status code
+  rescue_from ActionController::ParameterMissing do |e|
+    errors = { e.param => ["is required"] }
+    render json: errors, status: :bad_request
+  end
+
   def find_repo
-    full_name = [params[:owner], params[:name]].join("/")
+    full_name = params.require(%i[owner name]).join("/")
```

Comment:
Without the `params.require` here we get the `ActiveRecord::RecordNotFound` error raised after running a sql query with a bad `full_name` parameter. I think this is probably a slight misuse of strong params so let me know if there is a better alternative.